docker/errors.go

```diff
@@ -59,3 +63,144 @@ func registryHTTPResponseToError(res *http.Response) error {
 	}
 	return err
 }
+
+// Code below is taken from https://github.com/distribution/distribution/blob/a4d9db5a884b70be0c96dd6a7a9dbef4f2798c51/registry/client/errors.go
```

Comment:
@vrothberg and @mtrmac 's call, but this almost seems like this new code should be in a separate file.  I"m going back and forth on it.

Reply:
I concur. It should be a separate file.

In a second commit, we should also unexport the API and symbols.

Reply:
+1 to both, if this would end up happening.